src/pybamm/solvers/solution.py

```diff
@@ -460,9 +460,20 @@ def first_state(self):
         else:
             all_yps = self.all_yps[0][:, :1]
 
+        # check if all_ys is empty (i.e. if IDAKLU solver + output_variables used)
+        if isinstance(self.all_ys[-1], casadi.DM):
+            empty_ys = False
+        else:
+            empty_ys = self.all_ys[0].shape[0] == 0
+
```

Comment:
Can we pass this information from the solver and make it a `Solution` property? It would be good to check the root cause instead of a consequence.

Reply:
Added a 'variables_returned' flag to the Solution class to use for this